### DISCUSSION TITLE
Disable redis?

### DISCUSSION THREAD

**Discussion by joobei** [mastodon/mastodon#28515] [General]
### Pitch

From what I understand redis is meant for caching and to reduce server load for stuff that is frequently accessed. What if somebody is running a mastodon instance with just a single user on a computer with limited resources (like a Raspberry Pi). Would it make sense to disable redis and just rely on postgresql thus saving memory on the Rpi? Is it technically feasible?

### Motivation

Save memory on weak computers and reduce overall mastodon setup and maintenance complexity.

**Comment by vmstan** [comment]
It does do caching but it also holds all of the data for the background tasks and jobs that have to do be done by Sidekiq.

**Comment by viktorix** [comment]
With Redis no longer open source, maybe it's time to replace it with KeyDB?

**Reply by jgaskins** [reply]
The application doesn't control whether you use Redis or KeyDB since they're API-compatible with each other. You can choose Redis or KeyDB when you deploy it.

**Reply by soupglasses** [reply]
Issue opened here: https://github.com/mastodon/mastodon/issues/29729
